plugins/module_utils/dnac.py

```diff
@@ -202,6 +210,83 @@ def dnac_argument_spec():
     )
     return argument_spec
 
+def validate_str(item, param_spec, param_name, invalid_params):
+    item = validation.check_type_str(item)
+    if param_spec.get("length_max"):
+        if 1 <= len(item) <= param_spec.get("length_max"):
+            return item
+        else:
+            invalid_params.append(
+                "{0}:{1} : The string exceeds the allowed "
+                "range of max {2} char".format(
+                    param_name, item, param_spec.get("length_max")
+                )
+            )
+    return item
+
+def validate_int(item, param_spec, param_name, invalid_params):
+    item = validation.check_type_int(item)
+    min_value = 1
+    if param_spec.get("range_min") is not None:
+        min_value = param_spec.get("range_min")
+    if param_spec.get("range_max"):
+        if min_value <= item <= param_spec.get("range_max"):
+            return item
+        else:
+            invalid_params.append(
+                "{0}:{1} : The item exceeds the allowed "
+                "range of max {2}".format(
+                    param_name, item, param_spec.get("range_max")
+                )
+            )
+    return item
+
+def validate_bool(item, param_spec, param_name, invalid_params):
+    return validation.check_type_bool(item)
+
+def validate_list(item, param_spec, param_name, invalid_params):
+    try:
+        if param_spec.get("type") == type(item).__name__:
+            if param_spec:
```

Comment:
No need of this check.. If param_spec is already there and get the type before we check it.. 
As we have lot more statements in this if block, we can check with not equal and return in the beginning itself.